crates/ide-completion/Cargo.toml

```diff
@@ -17,6 +17,7 @@ itertools = "0.10.5"
 
 once_cell = "1.17.0"
 smallvec.workspace = true
+unicode-ident = "1.0.0"
```

Comment:
I'm unsure about this. `unicode-ident` is currently not a runtime dependency (it's only used as a helper in various proc macros) and it comes with some memory overhead. 10kB in static storage (see its [readme]) doesn't sound too bad (I mean, we use much larger memory in other places :sweat_smile:) but it'd be better if we could avoid it.

Assuming that people don't normally use non-ascii doc aliases, I'm wondering if we get away without it. I'd like to wait for @Veykril on this.

[readme]: https://github.com/dtolnay/unicode-ident#comparison-of-performance

Reply:
We could probably simplify it to exclude aliases containing spacing or common punctuation if the overhead is too much

Reply:
I think just doing a simple "starts with `char::is_alphabetic` and continues with `char::is_alphanumeric` or `_`" should probably suffice?